extensions/amp-animation/0.1/web-animations.js

```diff
@@ -84,18 +86,194 @@ const SERVICE_PROPS = {
+  }
+
+  /**
+   * @return {!WebAnimationPlayState}
+   */
+  getPlayState() {
+  }
+
+  /**
+   * @param {function(!WebAnimationPlayState)} unusedHandler
+   * @return {!UnlistenDef}
+   */
+  onPlayStateChanged(unusedHandler) {
+  }
+
+  /**
+  * Initializes the players but does not change the state.
+   */
+  init() {
+  }
+
+  /**
+   * Initializes the players if not already initialized,
+   * and starts playing the animations.
+   */
+  start() {
+  }
+
+  /**
+   */
+  pause() {
+  }
+
+  /**
+   */
+  resume() {
+  }
+
+  /**
+   */
+  reverse() {
+  }
+
+  /**
+   * @param {time} unusedTime
+   */
+  seekTo(unusedTime) {
+  }
+
+  /**
+   * Seeks to a relative position within the animation timeline given a
+   * percentage (0 to 1 number).
+   * @param {number} unusedPercent between 0 and 1
+   */
+  seekToPercent(unusedPercent) {
+  }
+
+  /**
+   */
+  finish() {
+  }
+
+  /**
+   */
+  cancel() {
+  }
+
+  /**
+   * @param {!WebAnimationPlayState} unusedPlayState
+   * @private
+   */
+  setPlayState_(unusedPlayState) {
+  }
+}
+
+/**
+ */
+export class AnimationWorkletRunner extends AnimationRunner {
 
-    /** @private {?Array<!Animation>} */
+  /**
+   * @param {!Window} win
+   * @param {!Array<!InternalWebAnimationRequestDef>} requests
+   */
+  constructor(win, requests) {
+    super(requests);
+
+    /** @const @private */
+    this.win_ = win;
+
+    /** @protected {?Array<!WorkletAnimation>} */
     this.players_ = null;
+  }
+
+  /**
+   * @return {string}
+   */
+  createCodeBlob_() {
+
+    return `
+    registerAnimator('anim${++animIdCounter}', class {
+      animate(currentTime, effect) {
+        // TODO: Do some work with \`viewport-margins\` here.
+        if (currentTime == NaN) {
+          return;
+        }
+        effect.localTime = currentTime;
+      }
+    });
+    `;
+  }
+
+  /**
+  * @override
+  * Initializes the players but does not change the state.
+   */
+  init() {
+    this.requests_.map(request => {
+      // Apply vars.
+      if (request.vars) {
+        setStyles(request.target,
+            assertDoesNotContainDisplay(request.vars));
+      }
+      // TODO(nainar): This switches all animations to AnimationWorklet.
+      // Limit only to Scroll based animations for now.
+      CSS.animationWorklet.addModule(
+          URL.createObjectURL(new Blob([this.createCodeBlob_()],
+              {type: 'text/javascript'}))).then(() => {
+        const scrollSource = this.win_.document./*OK*/scrollingElement;
+        const scrollTimeline = new this.win_.ScrollTimeline({
+          scrollSource,
+          orientation: 'block',
+          timeRange: request.timing.duration,
+        });
+        const keyframeEffect = new KeyframeEffect(request.target,
+            request.keyframes, request.timing);
+        const player = new this.win_.WorkletAnimation(`anim${animIdCounter}`,
+            [keyframeEffect],
+            scrollTimeline);
+        player.play();
+        this.players_ = player;
```

Comment:
should this be `this.players_.push(player)` ?

Reply:
Done